core/signeddata.go

```diff
@@ -34,6 +34,7 @@ var (
 	_ SignedData = Signature{}
 	_ SignedData = SignedVoluntaryExit{}
 	_ SignedData = VersionedSignedBlindedBeaconBlock{}
+	_ SignedData = SignedValidatorRegistration{} // name in the builder spec SignedValidatorRegistrationV1 or SignedBuilderRegistration
```

Comment:
Not sure what naming convention you would like for this section of the codebase ie builderRegistration or validatorRegistration 

within go-eth2-client its using the spec name validatorRegistration